src/main/java/ca/corefacility/bioinformatics/irida/ria/web/ajax/projects/ProjectSamplesAjaxController.java

```diff
@@ -49,6 +71,126 @@ public ResponseEntity<SampleNameValidationResponse> validateNewSampleName(@Reque
+			return ResponseEntity.status(HttpStatus.BAD_REQUEST).body(new AjaxErrorResponse(e.getMessage()));
+		}
+	}
+
+	/**
+	 * Remove 1 or more samples from a project.
+	 *
+	 * @param projectId Identifier for the project
+	 * @param request   All information about the samples to remove
+	 * @return result of the removal
+	 */
+	@DeleteMapping("/remove")
+	public ResponseEntity<AjaxResponse> removeSamplesFromProject(@PathVariable long projectId,
+			@RequestBody RemoveSamplesRequest request) {
+		String result = uiSampleService.removeSamplesFromProject(projectId, request.getSampleIds());
+		return ResponseEntity.ok(new AjaxSuccessResponse(result));
+	}
+
+	/**
+	 * Download a zipped file containing sequencing reads for a list of samples.
+	 *
+	 * @param projectId Identifier for a project
+	 * @param request   Details about the download request including a list of sample identifiers
+	 * @param response  {@link HttpServletResponse}
+	 * @return Zipped file of sequence files
+	 */
+	@PostMapping("/download")
+	public ResponseEntity<StreamingResponseBody> downloadSamples(@PathVariable long projectId,
+			@RequestBody DownloadRequest request, HttpServletResponse response) {
+		return ResponseEntity.ok(uiSampleService.downloadSamples(projectId, request.getSampleIds(), response));
+	}
+
+	/**
+	 * Export the current state of the Project Samples table as either a CSV or Excel file.
+	 *
+	 * @param projectId Identifier for the current project
+	 * @param type      Type of file to export (CSV or Excel)
+	 * @param request   Current state of the samples table
+	 * @param response  {@link HttpServletResponse}
+	 * @param locale    current users {@link Locale}
+	 * @throws IOException Thrown if issue export the file
+	 */
+	@PostMapping("/export")
+	public void downloadSamplesSpreadsheet(@PathVariable long projectId, @RequestParam String type,
+			@RequestBody ProjectSamplesTableRequest request, HttpServletResponse response, Locale locale)
+			throws IOException {
+		uiSampleService.downloadSamplesSpreadsheet(projectId, type, request, response, locale);
+	}
+
+	/**
+	 * Get a list of all {@link Sample} identifiers within a specific project
+	 *
+	 * @param id Identifier for a Project
+	 * @return {@link List} of {@link Sample} identifiers
+	 */
+	@GetMapping("/identifiers")
+	public List<Long> getSampleIdsForProject(@RequestParam Long id) {
+		return uiSampleService.getSampleIdsForProject(id);
+	}
+
+	/**
+	 * Share / Move samples between projects
+	 *
+	 * @param request {@link ShareSamplesRequest} details about the samples to share
+	 * @param locale  current users {@link Locale}
+	 * @return Outcome of the share/move
+	 */
+	@PostMapping("/share")
+	public ResponseEntity<AjaxResponse> shareSamplesWithProject(@RequestBody ShareSamplesRequest request,
+			Locale locale) {
+		try {
+			uiSampleService.shareSamplesWithProject(request, locale);
+			return ResponseEntity.ok(new AjaxSuccessResponse(""));
+		} catch (Exception e) {
```

Comment:
Please change this to a specific exception instead of the generic `Exception` as that will capture a lot more and potentially leak sensitive info.

Reply:
Updated in a54773bb40f123d5cf6ccd4d62fa7f71d39b8b11